nni/experiment/config/training_services/local.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""
+Configuration for local training service.
+
+Check the reference_ for explaination of each field.
+
+You man also want to check `local training service doc`_.
+
+.. _reference: https://nni.readthedocs.io/en/stable/reference/experiment_config.html
+
+.. _local training service doc: https://nni.readthedocs.io/en/latest/TrainingService/LocalMode.html
+
+"""
+
+__all__ = ['LocalConfig']
+
+from dataclasses import dataclass
+from typing import List, Optional, Union
+
+from ..training_service import TrainingServiceConfig
+from ..utils import canonical_gpu_indices
+
+@dataclass(init=False)
+class LocalConfig(TrainingServiceConfig):
+    platform: str = 'local'
+    use_active_gpu: Optional[bool] = None
+    max_trial_number_per_gpu: int = 1
+    gpu_indices: Union[List[int], int, str, None] = None
+    reuse_mode: bool = False
+
+    def _canonicalize(self, parents):
+        super()._canonicalize(parents)
+        self.gpu_indices = canonical_gpu_indices(self.gpu_indices)
+        self.nni_manager_ip = None
```

Comment:
do we have `nni_manager_ip` in the training service config before?

Reply:
`nni_manager_ip`, `trial_command`, `trial_gpu_number`, etc were only in the top level config and not in training service configs.
But it was decided several releases ago, according to discussion, that training service config should contain all things needed by training service. And then these fields in top level config will work as "shortcuts" for convenience and for compatibility.